zebra-state/src/service/finalized_state/disk_format/tests/prop.rs

```diff
@@ -361,6 +362,16 @@ fn roundtrip_sapling_tree_root() {
     proptest!(|(val in any::<sapling::tree::Root>())| assert_value_properties(val));
 }
 
+#[test]
+fn roundtrip_sapling_subtree_data() {
+    let _init_guard = zebra_test::init();
+
+    proptest!(|(mut val in any::<NoteCommitmentSubtreeData<sapling::tree::Node>>())| {
+        val.end = val.end.clamp(Height(0), MAX_ON_DISK_HEIGHT);
```

Comment:
This might not do what you expect, because almost all the tests will be checking an `end` of `MAX_ON_DISK_HEIGHT`.

This issue is documented here, but for a string regex, rather than a clamp:
> In general, the ideal solution is to find a way to take a seed value and then use prop_map to transform it into the desired, irregular domain.
> This kind of works, but it has problems ... [it] also results in a pretty odd distribution of values. 

https://proptest-rs.github.io/proptest/proptest/tutorial/filtering.html
https://proptest-rs.github.io/proptest/proptest/tutorial/transforming-strategies.html

Instead, transform the input range so its distribution is roughly uniform:
```suggestion
        val.end.0 %= (MAX_ON_DISK_HEIGHT.0 + 1);
```

Reply:
This is now #7443